packages/core/core/src/config.ts

```diff
@@ -84,15 +84,7 @@ export class Config {
 
   events: Events;
 
-  embarkConfig: EmbarkConfig = {
-    contracts: [],
-    config: '',
-    versions: {
-      solc: ''
-    },
-    generationDir: '',
-    migrations: ''
-  };
```

Comment:
These changes here caused a test to fail.

Namely, these two: https://github.com/embarklabs/embark/blob/e29a93bde5d7ceb86141ef3500fe46404dff6e50/packages/embark/src/test/config.js#L205-L256

Here's why:

- The tests call `loadContractsConfigFiles()` individually (https://github.com/embarklabs/embark/blob/e29a93bde5d7ceb86141ef3500fe46404dff6e50/packages/embark/src/test/config.js#L206)
- This function relies on `embarkConfig.versions` to determine the solc compiler configuration: https://github.com/embarklabs/embark/blob/e29a93bde5d7ceb86141ef3500fe46404dff6e50/packages/core/core/src/config.ts#L438
- It then uses `embarkConfig.versions` to potentially override its own default: https://github.com/embarklabs/embark/blob/e29a93bde5d7ceb86141ef3500fe46404dff6e50/packages/core/core/src/configDefaults.ts#L46
- Since `versions.solc` is set here in `embarkConfig`, it'll override `getContractDefaults()` defaults and therefore no longer match the expectations in the tests. This is not an issue when `embarkConfig: any = {};`

^ This is rarely a problem in practice because we never call `loadContractsConfigFile()` individually. It usually gets called via `loadContractFiles()` which loads the embarkConfig from the dapp in which `versions` will be defined: https://github.com/embarklabs/embark/blob/e29a93bde5d7ceb86141ef3500fe46404dff6e50/packages/core/core/src/config.ts#L173

However, it will also call `loadEmbarkConfigFile()` which (ironically) doesn't actually load a file, but also comes with its own embarkConfig defaults which are merged with the loaded config: https://github.com/embarklabs/embark/blob/e29a93bde5d7ceb86141ef3500fe46404dff6e50/packages/core/core/src/config.ts#L620-L630

At this point it's clear that we need a single place for embarkConfig defaults which can then be relied on.

I was also considering making `embarkConfig.version` optional to make the test pass, but then TypeScript will rightfully complain in the coverage plugin that `embarkConfig.versions.solc` is potentially not defined, in which case we need a default again anyways.